executable.go

```diff
@@ -239,15 +211,56 @@ func _Execute( // nolint
 		}
 
 		node._DecreaseBackoff()
-		switch shouldRetry(logID, request, resp) {
+
+		// Check if the response is a querry or a transaction
+		tx, isTransaction := request.(*Transaction)
+		txLogger := client.logger
+		if isTransaction {
+			if tx.logLevel != nil {
+				txLogger = client.logger.SetLevel(*tx.logLevel)
+			}
+		} else {
+			q, _ := request.(*Query)
+			if q.logLevel != nil {
+				txLogger = client.logger.SetLevel(*q.logLevel)
+			}
+		}
```

Comment:
Maybe this should happen outside of the big for loop and it could be in a utility function to increase readibility

Reply:
refactored